### PR TITLE
Fix #1526: Implement java.lang.Thread.{get,set}Name.

### LINKED ISSUE #1526: java.lang.Thread.{get,set}Name()
Another one needed by ScalaTest, with an even simpler fix.
Note: on the JVM, it seems the default name for the main thread is `main`.

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

File: test-suite/src/test/scala/org/scalajs/testsuite/javalib/ThreadTest.scala [hunk 1]
```diff
@@ -14,6 +14,15 @@ import org.scalajs.jasminetest.JasmineTest
 object ThreadTest extends JasmineTest {
 
   describe("java.lang.Thread") {
+    it("getName and setName") {
+      val t = Thread.currentThread()
+      expect(t.getName).toBe("main") // default name of the main thread
+      t.setName("foo")
+      expect(t.getName).toBe("foo")
+      t.setName("main") // don't pollute the rest of the world with this test
```
Comment (gzm0) [inline test-suite/src/test/scala/org/scalajs/testsuite/javalib/ThreadTest.scala]: In a finally block, in case the `expect` above fails?

**Comment by sjrd** [comment]
Updated with the try-finally.

**Comment by sjrd** [comment]
retest this please

**Comment by gzm0** [comment]
LGTM

**Comment by sjrd** [comment]
Rebuilding one failed task: https://scala-webapps.epfl.ch/jenkins/job/scalajs-task-worker/31462/

**Comment by sjrd** [comment]
Rebuild passed.

**Pull request merged**